### PR TITLE
use small model to run integ test

### PR DESCRIPTION
**Pull request by ylwu-amzn**
Signed-off-by: Yaliang Wu <ylwu@amazon.com>

### Description
[Describe what this change achieves]
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [ ] All tests pass
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
- [ ] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java [hunk 1]
```diff
@@ -136,7 +139,7 @@ private static void validateMLInput(Input input) {
         }
     }
 
-    private static void validateInput(Input input) {
+    private void validateInput(Input input) {
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java:142]: I assume we are removing `static` to make it easy in writing unit test, right?

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java [hunk 2]
```diff
@@ -55,10 +53,11 @@ public class ModelHelper {
     public void downloadAndSplit(String modelId, String modelName, String version, String url, ActionListener<Map<String, Object>> listener) {
         try {
             AccessController.doPrivileged((PrivilegedExceptionAction<Void>) () -> {
-                Path modelUploadPath = getUploadModelPath(modelId, modelName, version);
+                Path modelUploadPath = mlEngine.getUploadModelPath(modelId, modelName, version);
                 String modelPath = modelUploadPath +".zip";
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java:57]: may be we can put `.zip` as a static variable?

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java [hunk 3]
```diff
@@ -55,10 +53,11 @@ public class ModelHelper {
...
                 Path modelPartsPath = modelUploadPath.resolve("chunks");
                 File modelZipFile = new File(modelPath);
+                log.debug("download model to file {}", modelZipFile.getAbsolutePath());
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java:60]: What about: "Starting to download the model at: {}"

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/utils/FileUtils.java [hunk 4]
```diff
@@ -108,13 +111,14 @@ public static void mergeFiles(Queue<File> files, File mergedFile) {
 
                     write(fileContent, mergedFile, true);
                 }
+            } catch (IOException e) {
+                log.error("Failed to merge file " + f.getAbsolutePath() + " to " + mergedFile.getAbsolutePath(), e);
```
Comment (dhrubo-os) [inline ml-algorithms/src/main/java/org/opensearch/ml/engine/utils/FileUtils.java:115]: what about: `log.error("Failed to merge files from " + f.getAbsolutePath() + " to " + mergedFile.getAbsolutePath(), e);`

File: plugin/src/main/java/org/opensearch/ml/action/forward/TransportForwardAction.java [hunk 5]
```diff
@@ -36,59 +36,36 @@
...
         MLTaskManager mlTaskManager,
-        ClusterService clusterService,
-        ThreadPool threadPool,
         Client client,
-        NamedXContentRegistry xContentRegistry,
-        MLTaskDispatcher mlTaskDispatcher,
-        MLIndicesHandler mlIndicesHandler,
         MLModelManager mlModelManager,
-        DiscoveryNodeHelper nodeFilter
+        DiscoveryNodeHelper nodeHel
```
Comment (dhrubo-os) [inline plugin/src/main/java/org/opensearch/ml/action/forward/TransportForwardAction.java:62]: `nodeHelper`?

**Review by dhrubo-os — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java [hunk 6]
```diff
@@ -172,111 +174,107 @@ public void uploadMLModel(MLUploadInput uploadInput, MLTask mlTask) {
...
+                indexModelMetaRequest.source(mlModelMeta.toXContent(XContentBuilder.builder(JSON.xContent()), EMPTY_PARAMS));
+                indexModelMetaRequest.setRefreshPolicy(WriteRequest.RefreshPolicy.IMMEDIATE);
+                // create model meta doc
+                ActionListener<IndexResponse> listener = ActionListener.wrap(modelMetaRes -> {
+                    String modelId = modelMetaRes.getId();
+                    mlTask.setModelId(modelId);
+                    log.info("create new model meta doc {} for upload task {}", modelId, taskId);
+                    modelHelper.downloadAndSplit(modelId, modelName, version, uploadInput.getUrl(), ActionListener.wrap(result -> {
+                        Long modelSizeInBytes = (Long) result.get(MODEL_SIZE_IN_BYTES);
+                        if (modelSizeInBytes / 1024 / 1024 / 1024 >= 4L) {
```
[the remark this answers is not in the document]
Comment (jngz-es) [reply plugin/src/main/java/org/opensearch/ml/model/MLModelManager.java:206]: From computing perspective, it's fine, as the compiler will optimize "/ 1024" to right shift operation. It is just style, how about having a constant like 4 * 1024 * 1024 * 1024?

**Review by jngz-es — APPROVED** [review]

File: plugin/src/main/java/org/opensearch/ml/action/forward/TransportForwardAction.java (hunk 5 above)
Comment (ylwu-amzn) [reply plugin/src/main/java/org/opensearch/ml/action/forward/TransportForwardAction.java:62]: Good catch, will fix in next PR

File: ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java (hunk 1 above)
Comment (ylwu-amzn) [reply ml-algorithms/src/main/java/org/opensearch/ml/engine/MLEngine.java:142]: Yes, will easier for both unit test and integration test.

**Pull request merged**